test/backup.py

```diff
@@ -123,19 +123,29 @@ def _check_data(self, t, count, msg):
         logging.exception('exception waiting for data to replicate')
       timeout = utils.wait_step(msg, timeout)
 
-  def _restore(self, t):
+  def _restore(self, t, tablet_type='replica'):
     """Erase mysql/tablet dir, then start tablet with restore enabled."""
     self._reset_tablet_dir(t)
     t.start_vttablet(wait_for_state='SERVING',
-                     init_tablet_type='replica',
+                     init_tablet_type=tablet_type,
                      init_keyspace='test_keyspace',
                      init_shard='0',
                      supports_backups=True)
 
+    # check semi-sync is enabled for replica, disabled for rdonly.
+    if tablet_type == 'replica':
+      t.check_db_var('rpl_semi_sync_slave_enabled', 'ON')
+      t.check_db_status('rpl_semi_sync_slave_status', 'ON')
+    else:
+      t.check_db_var('rpl_semi_sync_slave_enabled', 'OFF')
+      t.check_db_status('rpl_semi_sync_slave_status', 'OFF')
+
   def _reset_tablet_dir(self, t):
     """Stop mysql, delete everything including tablet dir, restart mysql."""
     utils.wait_procs([t.teardown_mysql()])
-    t.remove_tree()
+    # specify ignore_options because we want to delete the tree even
+    # if -k was specified on the command line.
```

Comment:
Can you explain what -k on the command line means? And is it command line to the test?